src/main/java/sc/fiji/bdvpg/bdv/projector/AccumulateMixedProjectorARGB.java

```diff
@@ -127,26 +124,38 @@ else if ( projectionMode.equals( "Avg" ))
 
 	}
 
-	private String getProjectionMode( int sourceIndex, Map< SourceAndConverter, Map< String, Object > > sourceAndConverterToMetadata )
+	private String[] getProjectionModes( ArrayList< Source< ? > > sources )
 	{
-		Source< ? > source = sourceList.get( sourceIndex );
-		if ( source instanceof TransformedSource )
-			source = (( TransformedSource )source).getWrappedSource();
+		final List< SourceAndConverter > sacs = SourceAndConverterServices.getSourceAndConverterService().getSourceAndConverters();
+		final String[] projectionModes = new String[ sources.size() ];
 
-		String projectionMode = "Sum";
+		int sourceIndex = 0;
 
-		for ( SourceAndConverter sac : sourceAndConverterToMetadata.keySet() )
+		for ( Source<?> source : sources )
 		{
-			if ( sac.getSpimSource().equals( source ) )
-			{
-				final Set< String > metadata = sourceAndConverterToMetadata.get( sac ).keySet();
-				if ( metadata.contains( "ProjectionMode" ) )
-				{
-					projectionMode = (String) sourceAndConverterToMetadata.get( sac ).get( "ProjectionMode" );
-				}
-			}
+			if ( source instanceof TransformedSource )
+				source = ( ( TransformedSource ) source ).getWrappedSource();
+
+			SourceAndConverter sac = SourceAndConverterServices.getSourceAndConverterService().getSourceAndConverterFromSource( source );
```

Comment:
This could fail because a `Source` can have different converters and thus exists in different flavors. One Source -> 2 SourceAndConverters, one with a gray LUT and one with a completely different LUT, for instance.

We have to think about that, but that's a major issue. I had to deal with this in the remove(SourceAndConverter, BdvHandle) function from the display service, you should have a look at it and then let's rediscuss this. Maybe there's no better way...

Reply:
Agreed! See conversations below.